### PR TITLE
Ignore TRAP after CHECK failure

### PR DESCRIPTION
**Pull request by backes**
This addresses https://crbug.com/1190759.

### CODE REVIEW

File: src/python/tests/core/crash_analysis/stack_parsing/stack_analyzer_data/v8_check_trap.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#
+# Fatal error in foo.cc, line 4711
+# Check failed: 3 == 2
+#
+AddressSanitizer:DEADLYSIGNAL
+=================================================================
+==1450490==ERROR: AddressSanitizer: TRAP on unknown address 0x000000000000 (pc 0x7ff87c68081b bp 0x7ffd59cb1bf0 sp 0x7ffd59cb1be0 T0)
+SUMMARY: AddressSanitizer: TRAP (/foobar/libv8_libbase.so+0x4d81b)
+==1450490==ABORTING
+
```
Comment (oliverchang) [inline src/python/tests/core/crash_analysis/stack_parsing/stack_analyzer_data/v8_check_trap.txt:10]: can you add some stackframes here for a more realistic test?

File: src/python/lib/clusterfuzz/stacktraces/__init__.py [hunk 2]
```diff
@@ -674,9 +674,10 @@ def parse(self, stacktrace: str) -> CrashInfo:
         continue
 
       # Sanitizer regular crash (includes ills, abrt, etc).
-      # However if we have a Go, Python, or Kernel crash, don't replace
-      # this with sanitizer signal handler failure.
+      # Certain crash types have precedence though; don't overwrite them.
       if (not state.crash_type.startswith('Kernel failure') and
+          not state.crash_type in
+            IGNORE_CRASH_TYPES_FOR_ABRT_BREAKPOINT_AND_ILLS and
```
Comment (oliverchang) [inline src/python/lib/clusterfuzz/stacktraces/__init__.py:680]: Can you modify line 422 onwards instead? Just need to add a new pattern for TRAPs.

```python
      # Ignore aborts, breakpoints and ills for asserts, check and dcheck
      # failures. These are intended, retain their original state.
      if (SAN_ABRT_REGEX.match(line) or SAN_BREAKPOINT_REGEX.match(line) or
          SAN_ILL_REGEX.match(line)):
        if state.crash_type in IGNORE_CRASH_TYPES_FOR_ABRT_BREAKPOINT_AND_ILLS:
          continue
```

**Review by oliverchang — CHANGES_REQUESTED** [review]
thanks!

File: src/python/tests/core/crash_analysis/stack_parsing/stack_analyzer_data/v8_check_trap.txt (hunk 1 above)
Comment (backes) [reply src/python/tests/core/crash_analysis/stack_parsing/stack_analyzer_data/v8_check_trap.txt:10]: Ack, I copied in a complete stack trace now.

File: src/python/lib/clusterfuzz/stacktraces/__init__.py (hunk 2 above)
Comment (backes) [reply src/python/lib/clusterfuzz/stacktraces/__init__.py:680]: Ah, didn't find that location. It indeed fits nicer there. Thanks!

**Review by oliverchang — APPROVED** [review]
nice, thanks!

**Comment by oliverchang** [comment]
lint error not caused by this PR. merging and fixing the lint in another PR.

**Pull request merged**